Lib/test/test_annotationlib.py

```diff
@@ -1205,6 +1211,159 @@ def test_annotations_to_string(self):
         )
 
 
+class A:
+    pass
+
+
+class ForwardRefTests(unittest.TestCase):
+    def test_forwardref_instance_type_error(self):
+        fr = ForwardRef('int')
+        with self.assertRaises(TypeError):
+            isinstance(42, fr)
+
+    def test_forwardref_subclass_type_error(self):
+        fr = ForwardRef('int')
+        with self.assertRaises(TypeError):
+            issubclass(int, fr)
+
+    def test_forwardref_only_str_arg(self):
+        with self.assertRaises(TypeError):
+            ForwardRef(1)  # only `str` type is allowed
+
+    def test_forward_equality(self):
+        fr = ForwardRef('int')
+        self.assertEqual(fr, ForwardRef('int'))
+        self.assertNotEqual(List['int'], List[int])
+        self.assertNotEqual(fr, ForwardRef('int', module=__name__))
+        frm = ForwardRef('int', module=__name__)
+        self.assertEqual(frm, ForwardRef('int', module=__name__))
+        self.assertNotEqual(frm, ForwardRef('int', module='__other_name__'))
+
+    def test_forward_equality_gth(self):
```

Comment:
since we're moving things around anyway, could we rename this test? It always took me a second or two to figure out what `gth` meant even when it was in `test_typing.py`, and the context is even less obvious in its new location

```suggestion
    def test_forward_equality_get_type_hints(self):
```